packages/agreement-process/src/services/agreementStampUtils.ts

```diff
@@ -35,7 +35,7 @@ export const suspendedByProducerStamp = (
   requesterOrgId: TenantId,
   destinationState: AgreementState,
   stamp: AgreementStamp,
-  delegateProducerId?: TenantId | undefined
+  delegateProducerId: TenantId | undefined
```

Comment:
By removing the question mark here I enforce that this param shall always be explicitly set to some value, either a Delegation or undefined. An example of what I proposed here: https://pagopa.atlassian.net/browse/PIN-5598

That's how I found the bug while implementing feature/capofila